src/validation/global_validator.rs

```diff
@@ -5,13 +5,25 @@ use crate::{
     typesystem::{DataTypeInformation, StructSource},
 };
 
+use super::Validators;
+
 /// Validator that does not check a dedicated file but rather
 /// uses the index to validate the project as a whole.
 /// It performs validations including:
 ///  - naming-conflicts
 ///  - <tbc>
 pub struct GlobalValidator {
-    pub diagnostics: Vec<Diagnostic>,
+    diagnostics: Vec<Diagnostic>,
+}
+
+impl Validators for GlobalValidator {
+    fn push_diagnostic(&mut self, diagnostic: Diagnostic) {
+        self.diagnostics.push(diagnostic);
+    }
+
+    fn get_diagnostics(&mut self) -> &mut Vec<Diagnostic> {
```

Comment:
I would rather you use std::mem::take in these situations, it's clearer that you are now consuming the content of the diagnostics. Simply saying get_diagnostics implies you are just using them.